client/my-sites/checkout/checkout-thank-you/index.tsx

```diff
@@ -574,7 +575,7 @@ export class CheckoutThankYou extends Component<
 		}
 
 		/** REFACTORED REDESIGN */
-		if ( isRefactoredForThankYouV2( this.props ) ) {
+		if ( this.isDataLoaded() && isRefactoredForThankYouV2( this.props ) ) {
```

Comment:
This ensures we only perform the `isRefactoredForThankYouV2()` check when data is loaded. Otherwise, we end up showing the generic page for a split second before showing the specific congrats page (e.g. plan only congrats page).

Reply:
👍